### PR TITLE
[CELEBORN-869][DOC] Document on Integrating Celeborn

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
As title.

### Why are the changes needed?
As title.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manual test.

### CODE REVIEW

**Comment by waitinfuture** [comment]
cc @pan3793 @cfmcgrady @AngersZhuuuu

File: docs/developers/integrate.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+# Integrating Celeborn
+## Overview
+The core components of Celeborn, i.e. `Master`, `Worker`, and `Client` are all engine irrelevant. Developers can
+integrate Celeborn with various engines or applications by using or extending Celeborn's `Client`, as the officially
+supported plugins for Apache Spark and Apache Flink, see [Spark Plugin](../../developers/spark) and 
+[FLink Plugin](../../developers/flink).
```
Comment (cfmcgrady) [inline docs/developers/integrate.md:24]:
```suggestion
[Flink Plugin](../../developers/flink).
```

File: docs/developers/integrate.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+number of partitions for [Lazy Register](../../developers/shuffleclient#lazy-shuffle-register).
+
+After the map task finishes, `ShuffleClient` should call `mapperEnd` to tell `LifecycleManager` that the map task
+finishes pushing its data:
+
+```java
+public abstract void mapperEnd(int shuffleId,
+        int mapId,
+        int attemptId,
+        int numMappers)
```
Comment (cfmcgrady) [inline docs/developers/integrate.md:144]:
```suggestion
public abstract void mapperEnd(
    int shuffleId,
    int mapId,
    int attempted,
    int numMappers)
```

File: docs/developers/integrate.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+For simplicity, to read the whole data from the partition, you can pass 0 and `Integer.MAX_VALUE` to `startMapIndex`
+and `endMapIndex`. This method will create an InputStream for the data, and guarantees no data lost and no
+duplicate reading, else exception will be thrown.
+
+## Step Five: Clean Up
+After the shuffle finishes, you can call `LifecycleManager.unregisterShuffle` to clean up resources related to the
+shuffle:
+
+```java
+lm.unregisterShuffle(0)
```
Comment (cfmcgrady) [inline docs/developers/integrate.md:175]:
```suggestion
lm.unregisterShuffle(0);
```

**Review by cfmcgrady — APPROVED** [review]

**Comment by waitinfuture** [comment]
Thanks, merging to main.

**Pull request closed without merging**